src/raft/raw_node.rs

```diff
@@ -300,6 +312,9 @@ impl<T: Storage> RawNode<T> {
            raft.raft_log.has_next_entries() {
             return true;
         }
+        if raft.read_state.index != INVALID_INDEX {
```

Comment:
I don't know, but if we have already been working for it, we can still send the PR.